cuda_core/tests/memory_ipc/test_workerpool.py

```diff
@@ -0,0 +1,105 @@
+
+# Global memory resources, set in children.
+g_mrs = None
+
+
+class TestIpcWorkerPoolUsingExport:
+    """
+    Test buffer sharing using export handles.
+
+    The memory resources need to be passed to subprocesses at startup. Buffers
+    are passed by their handles and reconstructed using the corresponding mr.
+    """
+
+    @staticmethod
+    def init_worker(mrs):
+        global g_mrs
+        g_mrs = mrs
+
+    def test_ipc_workerpool(self, ipc_device, ipc_memory_resource):
+        """Test IPC with a worker pool."""
+        device = ipc_device
+        mr = ipc_memory_resource
+        buffers = [mr.allocate(NBYTES) for _ in range(NTASKS)]
+        with multiprocessing.Pool(processes=NWORKERS, initializer=self.init_worker, initargs=([mr],)) as pool:
+            pool.starmap(self.process_buffer, [(0, buffer.get_ipc_descriptor()) for buffer in buffers])
+
+        for buffer in buffers:
+            IPCBufferTestHelper(device, buffer).verify_buffer(flipped=True)
+
+    def test_ipc_workerpool_multi_mr(self, ipc_device, ipc_memory_resource):
+        """Test IPC with a worker pool using multiple memory resources."""
+        device = ipc_device
+        options = DeviceMemoryResourceOptions(max_size=POOL_SIZE, ipc_enabled=True)
+        mrs = [ipc_memory_resource] + [DeviceMemoryResource(device, options=options) for _ in range(NMRS - 1)]
+        buffers = [mr.allocate(NBYTES) for mr, _ in zip(cycle(mrs), range(NTASKS))]
+        with multiprocessing.Pool(processes=NWORKERS, initializer=self.init_worker, initargs=(mrs,)) as pool:
+            pool.starmap(
+                self.process_buffer,
+                [(mrs.index(buffer.memory_resource), buffer.get_ipc_descriptor()) for buffer in buffers],
+            )
+
+        for buffer in buffers:
+            IPCBufferTestHelper(device, buffer).verify_buffer(flipped=True)
+
+    def process_buffer(self, mr_idx, buffer_desc):
+        device = Device()
+        buffer = Buffer.from_ipc_descriptor(g_mrs[mr_idx], buffer_desc)
+        IPCBufferTestHelper(device, buffer).fill_buffer(flipped=True)
+
+
+class TestIpcWorkerPool:
+    """
+    Test buffer sharing without using export handles.
+
+    The memory resources need to be passed to subprocesses at startup. Buffers
+    are serialized with the `uuid` of the corresponding mr, and the
+    import/export is handled automatically.
+    """
+
+    @staticmethod
+    def init_worker(mrs):
+        global g_mrs
+        g_mrs = mrs
+
+    def test_ipc_workerpool(self, ipc_device, ipc_memory_resource):
+        """Test IPC with a worker pool."""
+        device = ipc_device
+        mr = ipc_memory_resource
+        buffers = [mr.allocate(NBYTES) for _ in range(NTASKS)]
+        with multiprocessing.Pool(processes=NWORKERS, initializer=self.init_worker, initargs=([mr],)) as pool:
+            pool.map(self.process_buffer, buffers)
+
+        for buffer in buffers:
+            IPCBufferTestHelper(device, buffer).verify_buffer(flipped=True)
```

Comment:
I am staring the screen for a few mins, and I feel this test can be merged into the one below, by parametrizing  over `NMRS` (1 and 2). I think the test below supports both cases, and this one is redundant. WDYT?